optaweb-vehicle-routing-backend/src/main/java/org/optaweb/vehiclerouting/plugin/planner/domain/PlanningLocation.java

```diff
@@ -0,0 +1,141 @@
+
+
+    private BigDecimal latitude;
+    private BigDecimal longitude;
+    private String description;
+
+    public PlanningLocation() {
+    }
+
+    public PlanningLocation(long id, double latitude, double longitude) {
+        super(id);
+        this.latitude = BigDecimal.valueOf(latitude);
+        this.longitude = BigDecimal.valueOf(longitude);
+    }
+
+    public PlanningLocation(long id, BigDecimal latitude, BigDecimal longitude) {
+        super(id);
+        this.latitude = latitude;
+        this.longitude = longitude;
+    }
+
+    public PlanningLocation(long id, BigDecimal latitude, BigDecimal longitude, String description) {
+        super(id);
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.description = description;
+    }
+
+    public PlanningLocation(Location location) {
+        if(location.coordinates() != null) {
+            latitude = location.coordinates().latitude();
+            longitude = location. coordinates().longitude();
+        }
+        description = location.description();
+        id = location.id();
+    }
+
+
+
+    // Prefer Map over array or List because customers might be added and removed in real-time planning.
+    private Map<PlanningLocation, Double> travelDistanceMap;
+
+    public BigDecimal getLatitude() {
+        return latitude;
+    }
+
+    public void setLatitude(BigDecimal latitude) {
+        this.latitude = latitude;
+    }
+
+    public BigDecimal getLongitude() {
+        return longitude;
+    }
+
+    public void setLongitude(BigDecimal longitude) {
+        this.longitude = longitude;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    public Map<PlanningLocation, Double> getTravelDistanceMap() {
+        return travelDistanceMap;
+    }
+
+    public void setTravelDistanceMap(Map<PlanningLocation, Double> travelDistanceMap) {
+        this.travelDistanceMap = travelDistanceMap;
+    }
+
+    /**
+     * Get distance to the given location
+     * @param location
+     * @return
+     */
+    public long getDistanceTo(PlanningLocation location) {
+        if (this == location) {
+            return 0L;
+        }
+        double distance = travelDistanceMap.get(location);
+        // Multiplied by 1000 to avoid floating point arithmetic rounding errors
+        return (long) (distance * 1000.0 + 0.5);
+    }
+
+    /**
+     * The angle relative to the direction EAST.
+     * @param location never null
+     * @return in Cartesian coordinates
+     */
+    public double getAngle(PlanningLocation location) {
+        // Euclidean distance (Pythagorean theorem) - not correct when the surface is a sphere
+        double latitudeDifference = latitude.subtract(latitude).longValue();
+        double longitudeDifference = longitude.subtract(longitude).longValue();
```

Comment:
This is `x - x` which is always zero. Breaks weight comparison based on angle. CH no longer produces pizza-like solution and finishes with much worse score. Will be fixed in #283.